### PR TITLE
Add error check in service_lb_controller

### PR DESCRIPTION
**Pull request by TaoZou1**
Add more error check in service_lb_controller

### CODE REVIEW

File: pkg/controllers/service/service_lb_controller.go [hunk 1]
```diff
@@ -55,24 +61,31 @@ func (r *ServiceLbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	}()
 
 	if err := r.Client.Get(ctx, req.NamespacedName, service); err != nil {
+		if client.IgnoreNotFound(err) == nil {
```
Comment (timdengyun) [inline pkg/controllers/service/service_lb_controller.go:64]: maybe we should use
if apierrors.IsNotFound(err) {

File: pkg/controllers/service/service_lb_controller.go [hunk 2]
```diff
@@ -55,24 +61,31 @@ func (r *ServiceLbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
...
+			log.Info("Not found LB service", "req", req.NamespacedName)
+			return ResultNormal, client.IgnoreNotFound(err)
```
Comment (timdengyun) [inline pkg/controllers/service/service_lb_controller.go:66]: Do we need yo requeue for this not found case?

File: pkg/controllers/service/service_lb_controller.go [hunk 3]
```diff
@@ -55,24 +61,31 @@ func (r *ServiceLbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
...
+			}
 		}
 	}
 
 	return ResultNormal, nil
 }
 
-func (r *ServiceLbReconciler) setServiceLbStatus(ctx context.Context, lbService *v1.Service) {
+func (r *ServiceLbReconciler) setServiceLbStatus(ctx context.Context, lbService *v1.Service) error {
 	ipMode := v1.LoadBalancerIPModeProxy
```
Comment (timdengyun) [inline pkg/controllers/service/service_lb_controller.go:89]: ipMode -> ipModeProxy

File: pkg/controllers/service/service_lb_controller.go [hunk 4]
```diff
@@ -55,24 +61,31 @@ func (r *ServiceLbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
...
+			return ResultNormal, client.IgnoreNotFound(err)
+		}
 		log.Error(err, "Failed to fetch LB service", "req", req.NamespacedName)
-		return ResultNormal, client.IgnoreNotFound(err)
+		return common.ResultRequeue, err
 	}
 
 	if service.Spec.Type == v1.ServiceTypeLoadBalancer {
-		log.Info("Reconciling LB service", "LBService", req.NamespacedName)
+		log.Info("Reconciling LB service", "LBService", req.NamespacedName, "version", service.ResourceVersion, "status", service.Status)
```
Comment (dantingl) [inline pkg/controllers/service/service_lb_controller.go:73]: put version and status in debug log level

File: pkg/controllers/service/service_lb_controller.go [hunk 5]
```diff
@@ -40,10 +40,16 @@ type ServiceLbReconciler struct {
...
+		r.Recorder.Event(lbService, v1.EventTypeNormal, common.ReasonSuccessfulUpdate, "LoadBalancer service has been successfully updated")
+		metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateSuccessTotal, MetricResType)
+		return nil
+	}
+	r.Recorder.Event(lbService, v1.EventTypeWarning, common.ReasonFailUpdate, "Failed to update LoadBalancer service")
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateFailTotal, MetricResType)
+	return err
 }
 
 func (r *ServiceLbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```
Comment (dantingl) [inline pkg/controllers/service/service_lb_controller.go:55]: Remove line 57?

File: pkg/controllers/service/service_lb_controller.go (hunk 2 above)
Comment (TaoZou1) [reply pkg/controllers/service/service_lb_controller.go:66]: not found means it's been deleted. no need to requeue it

File: pkg/controllers/service/service_lb_controller.go (hunk 1 above)
Comment (TaoZou1) [reply pkg/controllers/service/service_lb_controller.go:64]: done

File: pkg/controllers/service/service_lb_controller.go (hunk 3 above)
Comment (TaoZou1) [reply pkg/controllers/service/service_lb_controller.go:89]: it seems that ipMode is better. The value may be VIP

File: pkg/controllers/service/service_lb_controller.go (hunk 5 above)
Comment (TaoZou1) [reply pkg/controllers/service/service_lb_controller.go:55]: Done

File: pkg/controllers/service/service_lb_controller.go (hunk 4 above)
Comment (TaoZou1) [reply pkg/controllers/service/service_lb_controller.go:73]: Done

**Review by timdengyun — APPROVED** [review]
LGTM

**Review by timdengyun — APPROVED** [review]
LGTM

**Review by dantingl — APPROVED** [review]

**Pull request merged**